### DISCUSSION TITLE
tab-width ignored for C files

### DISCUSSION THREAD

**Discussion by tavianator** [helix-editor/helix#11646] [Troubleshooting] [answered]
### Summary

I want literal tab characters to be rendered with width 8.  But it seems they always get width 2.

### Reproduction Steps

```console
tavianator@graphene $ cat ~/.config/helix/languages.toml
[[languages]]
name = "c"
scope = "source.c"
injection-regex = "c"
file-types = ["c"]
indent = { tab-width = 8, unit = "\t" }
tavianator@graphene $ cat foo.c
int main(void) {
        return 0;
}
tavianator@graphene $ xxd foo.c
00000000: 696e 7420 6d61 696e 2876 6f69 6429 207b  int main(void) {
00000010: 0a09 7265 7475 726e 2030 3b0a 7d0a       ..return 0;.}.
tavianator@graphene $ helix foo.c
```

![Screenshot_20240906_135350](https://github.com/user-attachments/assets/35f6d426-2302-409e-8de9-ff32b6076613)

### Platform

Arch Linux

### Terminal Emulator

konsole 24.08.0

### Installation Method

Arch Repos

### Helix Version

helix 24.7

**Comment by the-mikedavis** [comment, marked as answer]
There's a typo in `~/.config/helix/languages.toml`: the header should be `[[language]]` instead of `[[languages]]`

**Reply by tavianator** [reply]
Argh thanks so much.  Might be worth a warning?

**Reply by the-mikedavis** [reply]
I don't think it would be straightforward to add warnings or errors on unused keys because of the way we read `languages.toml` right now. (We read it once for the `[[languages]]` and `[langauge-server]` config and differently for the `[[grammar]]` entries.) We have plans to change how the grammars are specified though which will change that and in the long run we want to change how language configuration is defined entirely (#8853)
